modules/Magento/EceTool/Console/Command/ConfigShowStoreUrlCommand.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\EceTool\Console\Command;
+
+use Magento\Framework\Console\Cli;
+use Magento\Framework\UrlInterface;
+use Magento\Store\Model\Store;
+use Magento\Store\Model\StoreManagerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Command for getting store ulr by store id.
+ */
+class ConfigShowStoreUrlCommand extends Command
+{
+    /**#@+
+     * Names of input arguments or options.
+     */
+    const INPUT_ARGUMENT_STORE_ID = 'store-id';
+    const INPUT_OPTION_ALL = 'all';
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        StoreManagerInterface $storeManager
+    ) {
+        parent::__construct();
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * @inheritdoc
+     * @since 101.0.0
+     */
+    protected function configure()
+    {
+        $this->addArgument(
+            self::INPUT_ARGUMENT_STORE_ID,
+            InputArgument::OPTIONAL,
+            'Store ID'
+        );
+        $this->addOption(
+            self::INPUT_OPTION_ALL,
+            'a',
+            InputOption::VALUE_OPTIONAL,
+            'Return urls for all stores'
+        );
+
+        $this->setName('config:show:store-url')
+            ->setDescription(
+                'Shows store base url for given id.'
+            );
+        parent::configure();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
```

Comment:
I would propose 3 ways to solve this.

1. Add this command into the core with patches
1. Soft-dependency via `suggest` on `magento-store`
    * If the class exists - ECE-Tools can use it
3. Create a separate module - helper

Reply:
1. The first variant can be declined as changes not so valuable for the core.
    Delivering some changes into core take much of time and efforts.
2. It will require to run Magento Application in ece-tools, we can't use the store module independently.